07-contributor-guide.Rmd

```diff
@@ -38,515 +46,253 @@ Following good software development and coding practices simplifies collaboratio
-must be in place to support this pace. FIMS code review will utilize
-tools available via [GitHub](https://github.com/features/code-review),
-which allows reviewers to analyze code changes, provide inline comments,
-and view change histories.
+2. Periodically rebase your feature branch with the branch you plan on merging into in the future by (a) fetching all the changes stored in GitHub for the branches in your clone and (b) rebasing your feature branch, e.g., here we rebase to origin/main because we eventually want to merge into main. We can rebase to any remote or local branch though. Be careful when rebasing if you have already pushed your feature branch to GitHub because rebasing will rewrite your commit hashes in your feature branch. 
 
-[The Google code review developer guide](https://google.github.io/eng-practices/review/) provides a useful set of guidelines for both reviewers and code authors.
+```{r eval = FALSE}
+$ git fetch -a
+$ git rebase origin/main
+```
 
-Below is a flowchart for the FIMS code review process. The author starts by submitting 
-a pull request (PR), ensuring documentation, tests, and CI checks are complete, 
-then propose a reviewer. The reviewer receives the review request and either 
-executes the review independently or pairs with another team representative 
-if assistance is needed. Based on the review, changes may be requested, which
-the author must address before approval. Once the PR is approved, the author 
-merges it into the main branch.
+3. While editing code, commit regularly following [commit messages](#commit-messages) guidelines by (a) adding the changed files to the que and (2) committing them to the feature branch with a thoughtful commit message. If you configure your editor for git you can remove the `-m` argument and use a text file to make a more meaningful commit message with multiple lines.
 
-```{r, echo=FALSE}
-DiagrammeR::mermaid("
-        graph TB
-        A[Author: Submit a pull request PR<br/><br/>- Documentation included<br/>- Tests implemented<br/>- CI checks passed]
-        B[Author: Propose a reviewer<br/><br/>- Default assignee: an OST rep for a quick review]
-        
-        A-->B
-        B-->C[Reviewer: Receive code review request]
-        C --> |No assistance needed| E[Reviewer: Execute code review]
-        C --> |Help wanted| D[Reviewer: Add an additional reviewer to the PR] 
-        D --> E
-        E --> |Changes needed| F[Reviewer: Request changes] 
-        F --> G[Author: Make changes]
-        G --> H[Reviewer: Approve PR]
-        E --> |No changes needed| H
-        H --> I[Author: Rebase and merge PR]
- 
-        style A text-align:left, fill:#FFF
-        style B fill:#FFF
-        style G fill:#FFF
-        style I fill:#FFF
-", height = 1000)
+```{r eval = FALSE}
+$ git add <filename>
+$ git commit -m "feat: 50 character description" "Longer description"
+```
 
+4. Push your committed changes to GitHub.
```

Comment:
Do we want to recommend squashing commits before pushing final changes to GitHub?

Reply:
I will add some language about squashing but just as additional guidance, i.e., not a requirement.

Reply:
```
4. Consider interactively rebasing your commits to squash similar commits
into a single commit or rearrange the order of your commits. The following
example is for the situation where you want to look at the last ten commits.
{r eval = FALSE}
$ git rebase -i head~10
```